charon/src/bin/charon-driver/translate/translate_types.rs

```diff
@@ -42,6 +42,45 @@ impl<'tcx, 'ctx> ItemTransCtx<'tcx, 'ctx> {
         }
     }
 
+    fn check_at_most_one_existential_trait_ref(
+        &self,
+        span: Span,
+        preds: &Vec<hax::Binder<hax::ExistentialPredicate>>,
+    ) -> Result<(), Error> {
+        let all_ex_trait_ref = preds
+            .iter()
+            .filter_map(|binded_pred| {
+                if let hax::ExistentialPredicate::Trait(trait_ref) = &binded_pred.value {
+                    Some(trait_ref)
+                } else {
+                    None
+                }
+            })
+            .collect::<Vec<_>>();
+        if all_ex_trait_ref.len() <= 1 {
+            return Ok(());
+        }
+        // prepare to print
+        // all: Trait1 + Trait 2 + ...
+        let all = all_ex_trait_ref
+            .iter()
+            .map(|x| {
+                self.t_ctx
+                    .tcx
+                    .def_path_str(x.def_id.as_rust_def_id().unwrap())
+            })
+            .collect::<Vec<String>>()
+            .join(" + ");
```

Comment:
`Itertools` has a helper for this:
```suggestion
            .format(" + ");
```